google_cloudsql_mysql/main.tf

```diff
@@ -0,0 +1,126 @@
+/**
+ * # cloudsql-mysql
+ * Creates CloudSQL MySQL Instance.
+ */
+
+locals {
+  default_database_name = "${var.application}-${var.realm}-${var.environment}-${var.instance_version}"
+  database_name         = coalesce(var.custom_database_name, local.default_database_name)
+  tier                  = coalesce(var.tier_override, "db-custom-${var.db_cpu}-${var.db_mem_gb * 1024}")
+
+  default_replica_name = "${local.database_name}-replica"
+  replica_name         = coalesce(var.custom_replica_name, local.default_replica_name)
+
+  ip_addresses = google_sql_database_instance.primary.ip_address
+
+  enable_ha = var.realm == "prod" ? true : false
+}
+
+resource "google_sql_database_instance" "primary" {
+  name             = local.database_name
+  region           = var.region
+  database_version = var.database_version
+
+  settings {
+    tier = local.tier
+
+    availability_type = var.availability_type
+
+    disk_type = "PD_SSD"
+    dynamic "database_flags" {
+      for_each = var.database_flags
```

Comment:
clarifying for myself - this is where MySQL RDS parameter groups would be migrated to, right? Thinking of [param groups like used here](https://github.com/mozilla-it/itse-apps-prod-1-infra/pull/65/files), and it isn't clear to me in the [terraform docs for google_sql_database_instance](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance) what flags are allowed or not.

Reply:
Maybe we could add a link to the [supported flags](https://cloud.google.com/sql/docs/mysql/flags#list-flags-mysql) in the variable description?

Reply:
added

Reply:
@jbuck you're certain the usage examples I pointed to are covered in those flags? that's basically what i'm asking, and your response doesn't seem clear if you just wanted to point to docs or actually knew the answer.

Reply:
@cmharlow yeah, here's `character_set_server` for example: https://cloud.google.com/sql/docs/mysql/flags#mysql-c (sorry, I can't link directly to it). The other variables aren't able to be set, which is why they're not listed there. I think the might inherit from `character_set_server` though?

Reply:
okay, so its unknown atm, but you're assuming it should just work somehow. will do some minor digging to confirm this.

Reply:
Confirmed you can't set unsupported (by CloudSQL) flags:

```
% gcloud sql instances patch --database-flags character_set_client=utf8mb4 --project moz-fx-balrog-nonprod-1841 balrog-nonprod-dev-v2
The following message will be used for the patch API method.
{"name": "balrog-nonprod-dev-v2", "project": "moz-fx-balrog-nonprod-1841", "settings": {"databaseFlags": [{"name": "character_set_client", "value": "utf8mb4"}]}}
WARNING: This patch modifies database flag values, which may require your instance to be restarted. Check the list of supported flags - https://cloud.google.com/sql/docs/mysql/flags - to see if your instance will be restarted when this patch is submitted.

Do you want to continue (Y/n)?  

ERROR: (gcloud.sql.instances.patch) HTTPError 404: The requested flag is either misspelled or unsupported by Cloud SQL.
```

Reply:
alright, looks like its a known problem & hole in the GCP cloudsql docs. Basically, if we're on mysql 5.7 or above, setting `character_set_server` should set the parameters / type for the other columns, but its not consistent in its application & may require us to run some SQL to manually change the column types post database creation.